src/vellum/workflows/workflows/event_filters.py

```diff
@@ -46,7 +46,10 @@ def root_workflow_event_filter(workflow_definition: Type["BaseWorkflow"], event:
     if event.parent.type != "WORKFLOW":
         return False
 
-    return event.parent.workflow_definition == CodeResourceDefinition.encode(workflow_definition)
```

Comment:
[nit] let's override the `__eq__` method in `CodeResourceDefinition` to ensure that `.name`s are equal and `.module`s are equal